src/main/java/com/ericsson/ei/subscription/SubscriptionHandler.java

```diff
@@ -68,57 +65,53 @@
     private RunSubscription runSubscription;
 
     /**
-     * The method takes a aggregatedObject as argument and fetches all the
-     * subscriber from the database in order to match the subscription
-     * conditions in a separate thread.
+     * The method takes a aggregatedObject as argument and fetches all the subscriber from the
+     * database in order to match the subscription conditions in a separate thread.
      *
      * @param aggregatedObject
      * @param id
      */
+    @Async("subscriptionHandlerExecutor")
     public void checkSubscriptionForObject(final String aggregatedObject,
-                                           final String id) {
-        Thread subscriptionThread = new Thread(() -> {
-            List<String> subscriptions = mongoDBHandler.getAllDocuments(
+            final String id) {
+        mongoDBHandler.init();
```

Comment:
I'm concerned about creating new connections to Mongo DB every time. Perhaps we should keep a single connection for this threadpool?